src/applications/edu-benefits/components/SchoolSelectField.jsx

```diff
@@ -84,21 +84,27 @@ export class SchoolSelectField extends React.Component {
 
   handleOptionClick = ({ address1, address2, address3, city, facilityCode, name, state, zip, country }) => {
     this.props.selectInstitution({ address1, address2, address3, city, facilityCode, name, state });
-    const backendFriendlyCountries = {
-      USA: 'United States'
+    const isDomesticAddress = country === 'USA';
+    let street2 = address2;
+    if (address3) {
+      street2 = `${street2} ${address3}`;
+    }
+    const address = {
+      country: isDomesticAddress ? 'United States' : country,
+      street: address1,
+      street2,
+      city,
+      state,
+      postalCode: zip,
     };
+    if (!isDomesticAddress) {
+      address.viaSearchTool = true;
+    }
     this.props.onChange({
       ...this.props.formData,
       name,
       'view:facilityCode': facilityCode,
-      address: {
-        country: backendFriendlyCountries[country] || country,
-        street: address1,
-        street2: address2,
-        city,
-        state,
-        postalCode: zip,
-      }
+      address
```

Comment:
Did we move away from submitting the facility code for these addresses?

Reply:
Yes. Lihan requested that the first day both you and Claire were gone. Since the search tool provided all of the required address info to the frontend it seemed doable.

Reply:
We really shouldn't be doing these kinds of data transformations in the client. If we can't submit exactly what we get back from the API then something should change on the server or the data.

Reply:
Fascinating point @rianfowler, looping @jholton for any additional thoughts from the backend perspective on this idea for organizing logic.